src/app/pages/categories/management/categories-editor/pages-categories-editor.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { checkInputError } from 'src/app/common/utils/forms/common-utils-forms-form-utils';
+import { WalletTransactionType } from 'src/app/domains/transactions/domains.transactions.constants';
+import { PagesTransactionCategoriesService } from '../../pages-transaction-categories.service';
+import { TransactionCategoryFull } from '../../transaction-category-full.model';
+import { TransactionCategory } from '../../transaction-category.model';
+import { ITransactionCategoryModalFormType } from './pages-categories-editor.types';
+
+@Component({
+  selector: 'app-pages-categories-editor',
+  templateUrl: './pages-categories-editor.component.html',
+  styleUrls: [ './pages-categories-editor.component.scss' ],
+})
+export class PagesCategoriesEditorComponent implements OnInit {
+  public selectTransactionsTypes: Record<string, WalletTransactionType> = {
+    'Income': WalletTransactionType.Income,
+    'Expense': WalletTransactionType.Expense,
+  };
+
+  public form!: FormGroup<ITransactionCategoryModalFormType>;
+
+  public transactionTypeDisabledLoading = true;
+
+  private category: TransactionCategory | null = null;
+  private fullCategory: TransactionCategoryFull | null = null;
+
+  public constructor(
+    private readonly pagesTransactionCategoriesService: PagesTransactionCategoriesService,
+    private readonly dialogRef: MatDialogRef<PagesCategoriesEditorComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: TransactionCategory,
+  ) { }
+
+  public get nameIsNotProvided(): boolean {
+    return checkInputError(this.form, 'name', 'required');
+  }
+
+  public get nameIsToLong(): boolean {
+    return checkInputError(this.form, 'name', 'maxlength');
+  }
+
+  public get typeAlreadyUsed(): boolean {
+    return Number(this.fullCategory?.financialTransactionsCounter) > 0;
+  }
+
+  public ngOnInit(): void {
+    this.category = this.data;
+
+    this.form = new FormGroup<ITransactionCategoryModalFormType>({
+      name: new FormControl(this.data.name, {
+        validators: [
+          Validators.required,
+          Validators.maxLength(20),
+        ],
+        nonNullable: true,
+      }),
+      type: new FormControl({ value: this.data.type, disabled: true }, {
```

Comment:
`disabled: true` -> this we can already set to final value -> first get data from BE (or in case of creating new category it will have already transactions count === 0) and then create form with proper `disabled` value